### PR TITLE
DOC: special.entr: add context

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
gh-11015
gh-11006
gh-3421

#### What does this implement/fix?
https://github.com/scipy/scipy/issues/3421#issuecomment-100654668 reported confusion about the `scipy.special` functions `entr`, `rel_entr`, and `kl_div`. This attempts to clarify that they are used in the context of convex programming and that `scipy.stats.entropy` should be used for information theory.

### CODE REVIEW

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -3110,13 +3110,30 @@ def add_newdoc(name, doc):
 
     See Also
     --------
-    kl_div, rel_entr
+    kl_div
+    rel_entr
+    scipy.stats.entropy
```
Comment (mdhaber) [inline scipy/special/_add_newdocs.py:3115]: Wanted to see how this renders differently.

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -7419,8 +7438,8 @@ def add_newdoc(name, doc):
 
     References
     ----------
-    .. [1] Grant, Boyd, and Ye, "CVX: Matlab Software for Disciplined Convex
-        Programming", http://cvxr.com/cvx/
+    .. [1] Grant, Boyd, and Ye, "The CVX Users' Guide",
+           http://web.cvxr.com/cvx/doc/CVX.pdf
```
Comment (mdhaber) [inline scipy/special/_add_newdocs.py:7442]: I'm concerned that user might think we mean the GPL-licensed code. @person142 LMK if the Users' Guide is not the reference you had in mind.

Comment (tupui) [reply scipy/special/_add_newdocs.py:7442]: Really not a fan of links like these that break all the time and we don't know about it. But if there is really no DOI we can refer to.

Comment (steppi) [reply scipy/special/_add_newdocs.py:7442]: One possibility is that we could cite Boyd’s Convex Optimization book, which does have a stable doi and also gives definitions for these functions. 

https://doi.org/10.1017/CBO9780511804441

File: scipy/special/_add_newdocs.py [hunk 3]
```diff
@@ -3110,13 +3110,31 @@ def add_newdoc(name, doc):
 
     See Also
     --------
-    kl_div, rel_entr
+    kl_div
+    rel_entr
+    scipy.stats.entropy
```
Comment (mdhaber) [inline scipy/special/_add_newdocs.py:3115]:
```suggestion
    kl_div, rel_entr, scipy.stats.entropy
```

File: scipy/special/_add_newdocs.py (hunk 2 above)
Comment (mdhaber) [reply scipy/special/_add_newdocs.py]: Done.

File: scipy/special/_add_newdocs.py [hunk 4]
```diff
@@ -3110,13 +3110,29 @@ def add_newdoc(name, doc):
...
+
+        \sum_{i = 1}^n \mathrm{entr}(p_i).
+
+    To compute the latter quantity, use `scipy.stats.entropy`.
+
+    References
+    ----------
+    .. [1] Boyd, Stephen and Lieven Vandenberghe. *Convex optimization*.
+           Cambridge University Press, 2004.
+           :doi:`https://doi.org/10.1017/CBO9780511804441`
```
Comment (tupui) [inline scipy/special/_add_newdocs.py:3135]: Just FYI when using `:doi:` you can omit the `https://doi.org/` part. Same as `:arxiv:`, it only needs the identifier.

**Review by tupui — APPROVED** [review]
LGTM, thanks Matt!

**Comment by mdhaber** [comment]
Thanks!

**Pull request merged**